### DISCUSSION TITLE
I started working on the wolf4sdl port

### DISCUSSION THREAD

**Discussion by mxlgv** [klange/toaruos#275] [Show and tell]
Having caught a million crashes and 1000 problems, I managed to partially launch it. While I was trying to run it, I found a bug in the SDL. (I'll write it off in an issue later). Now the sound does not work for some reason and the frame rate is somehow strange. I've come across this before, probably SDL_Delay() or GetTicks() don't work correctly.

![image](https://user-images.githubusercontent.com/51446645/231894658-efe68bb0-c5cf-4e0c-a409-44e412b75687.png)
![image](https://user-images.githubusercontent.com/51446645/231894714-85701fe8-8e8c-4adc-b3b4-71c986beb636.png)

**Comment by mxlgv** [comment]
@klange I noticed that you have not implemented a timer in SDL. Can you implement this or should I do it myself? I already have some work. Checked on Wolf3D the frame rate became adequate.

**Reply by klange** [reply]
There are options in [src/timer/unix/SDL_systimer.c](https://github.com/klange/SDL/blob/master/src/timer/unix/SDL_systimer.c) that should work. The reason they have not been set up is likely that none of the previous ports - sdlquake, the old Doom port, or the various demos - made use of these features.

If you would like to try to get the timer suite working, we can update the package. Please submit pull requests to the repo linked above.

**Comment by mxlgv** [comment]
Here are my work. The port itself and shareware resources are in one package: 
[wolf4sdl-test.tar.gz](https://github.com/klange/toaruos/files/11255612/wolf4sdl-test.tar.gz)

There is still no sound. It seems to be unfinished. Or it can only work in one thread.

**Reply by klange** [reply]
Not sure on the audio. The SDL backend should be working. Do you get sound from the other ports? Doom and Quake should work for effects, though music doesn't work in either. If you manage to get the OPL emulation in Wolf3D working, that would be useful in Doom as well...

**Reply by mxlgv** [reply]
Please open issue in SDL.  There are problems and tasks that I would like to add there.
